### PR TITLE
Improve RDKit package

### PR DESCRIPTION
**Pull request by RMeli**
Improve `rdkit` package adding some additional variants.

The `rdkit` version change is also included in #36270.

---

Tested on macOS:
```bash
spack install rdkit@2022_09_5+contrib+descriptors3d+freesasa~xyz2mol
```

`+xyz2mol` did not work on macOS because of https://github.com/rdkit/rdkit/issues/5559.

Tested on Ubuntu:
```bash
spack install rdkit@2022_09_5+contrib+descriptors3d+freesasa+xyz2mol
```

---

Patching `freesasa` was necessary to compile on Ubuntu, where I was getting this error:
```
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'V=1'

5 errors found in build log:
     180    ar: `u' modifier ignored since `D' is the default (see `U')
     181    ranlib libfreesasa.a
     182    /home/rmeli/spack/lib/spack/env/gcc/gcc  -I/home/rmeli/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.3.0/libxml2-2.10.3-raxyjdwc2tninyiq4a
            o7dyqyqcrtd6w4/include/libxml2 -I/home/rmeli/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.3.0/libiconv-1.17-cvxxsircwbwr54qq253lpyvbdsmew
            unl/include -g -O2   -o example example.o libfreesasa.a -ljson-c -L/home/rmeli/spack/opt/spack/linux-ubuntu22.04-zen4/gcc-11.3.0/libxml2-2
            .10.3-raxyjdwc2tninyiq4ao7dyqyqcrtd6w4/lib -lxml2 -lpthread -ldl -lm
     183    mv -f .deps/main.Tpo .deps/main.Po
     184    mv -f .deps/cif.Tpo .deps/cif.Po
     185    /home/rmeli/spack/lib/spack/env/gcc/g++ -std=gnu++14  -g -O2   -o freesasa main.o cif.o -lc++ libfreesasa.a -ljson-c -L/home/rmeli/spack/o
            pt/spack/linux-ubuntu22.04-zen4/gcc-11.3.0/libxml2-2.10.3-raxyjdwc2tninyiq4ao7dyqyqcrtd6w4/lib -lxml2 -lpthread -ldl -lm
  >> 186    /usr/bin/ld: cannot find -lc++: No such file or directory
  >> 187    collect2: error: ld returned 1 exit status
  >> 188    make[2]: *** [Makefile:513: freesasa] Error 1
     189    make[2]: Leaving directory '/tmp/rmeli/spack-stage/spack-stage-freesasa-2.1.2-r7zkl54jktgqdaluulsi4uhpcg4oigyt/spack-src/src'
  >> 190    make[1]: *** [Makefile:383: all-recursive] Error 1
     191    make[1]: Leaving directory '/tmp/rmeli/spack-stage/spack-stage-freesasa-2.1.2-r7zkl54jktgqdaluulsi4uhpcg4oigyt/spack-src'
  >> 192    make: *** [Makefile:322: all] Error 2
```

### CODE REVIEW

**Comment by RMeli** [comment]
Is there any blocker for this PR? @bvanessen

File: var/spack/repos/builtin/packages/freesasa/package.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ class Freesasa(AutotoolsPackage):
     depends_on("json-c", when="+json")
     depends_on("libxml2", when="+xml")
 
+    patch("libcpp.patch", when="@2.1.2")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/freesasa/package.py:35]: Can you a comment above the path explaining:
1. Why the patch is needed
2. What is the effect of the patch
3. Where the patch was taken from

?

File: var/spack/repos/builtin/packages/rdkit/package.py [hunk 2]
```diff
@@ -44,18 +44,59 @@ class Rdkit(CMakePackage):
...
+    variant("yaehmop", default=True, when="@2022_09_5:", description="Build YAeHMOP wrapper")
+    variant(
+        "xyz2mol", default=False, when="@2022_09_5:", description="Build support for RDKit xyz2mol"
+    )
+    variant(
+        "descriptors3d",
+        default=True,
+        when="@2022_09_5:",
+        description="Build 3D descriptors calculators",
+    )
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/rdkit/package.py]: I think you can group these under a:
```python
with when("@2022_09_5:"):
    variant(
        "python", default=True, description="Build standard Python wrappers"
    )
    ...
```

File: var/spack/repos/builtin/packages/rdkit/package.py [hunk 3]
```diff
@@ -44,18 +44,59 @@ class Rdkit(CMakePackage):
...
+    depends_on("maeparser", when="@2022_09_5: +maeparser")
+    depends_on("eigen@3:", when="@2022_09_5: +descriptors3d")
 
-    extends("python")
+    depends_on("python@3:", when="@:2021_09_5")
+    depends_on("python@3:", when="@2022_09_5: +python")
+    depends_on("py-numpy", when="@:2021_09_5")
+    depends_on("py-numpy", when="@2022_09_5: +python")
+    extends("python", when="@:2021_09_5")
+    extends("python", when="@2022_09_5: +python")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/rdkit/package.py]: Also these new dependencies may be grouped in the `when` context manager above

File: var/spack/repos/builtin/packages/freesasa/package.py (hunk 1 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/freesasa/package.py:35]: I did not take the patch from anywhere.

**Comment by RMeli** [comment]
Thanks @alalazo. I tested the changes on macOS 13.3.1 with

```bash
spack install rdkit@2022_09_5+contrib+descriptors3d+freesasa~xyz2mol
```
and
```bash
spack install rdkit@2021_09_5
```

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@bvanessen Is this fine with you?

**Pull request merged**